test/test_torch.py

```diff
@@ -10779,9 +10779,32 @@ def test_tensor_factories_empty(self, device):
         self.assertEqual((0,), torch.bartlett_window(0, periodic=False, device=device).shape)
         self.assertEqual((0,), torch.hamming_window(0, device=device).shape)
         self.assertEqual((0,), torch.hann_window(0, device=device).shape)
+        self.assertEqual((0,), torch.vandermonde_matrix(torch.tensor([]), 0, device=device).shape)
         self.assertEqual((1, 1, 0), torch.tensor([[[]]], device=device).shape)
         self.assertEqual((1, 1, 0), torch.as_tensor([[[]]], device=device).shape)
 
+    @unittest.skipIf(not TEST_NUMPY, 'Numpy not found')
+    @dtypes(torch.long, torch.double)
```

Comment:
What happens for bool, uint8, float, and complex64? 

We can still compare for closeness of the NumPy array and the tensor even if they're not the same dtype.

Reply:
issue with complex64, `at::cumprod` does not support complex64

> RuntimeError: "cumprod_out_cpu" not implemented for 'ComplexFloat'

add a check to prevent complex types as args for the vander function? Move away from using `at::cumprod` and add in a for loop?

Reply:
Cool. Add a check that when you try complex64 you get a RuntimeError. That way if the behavior changes we'll know to test it.